serveradmin/api/api.py

```diff
@@ -0,0 +1,34 @@
+from datetime import timedelta
+
+from django.utils import timezone
+
+from serveradmin.api.decorators import api_function
+from serveradmin.api.models import Lock
+
+
+@api_function(group='api')
+def lock(identifier, seconds=60):
+    """Mark identifier as in-use for n seconds
+
+    Allow clients to mark resources with an identifier as in-use so that
+    distributed clients can check whatever something is in use or not.
+
+    :param identifier: A unique identifier (e.g. 10.0.0.1)
+    :param seconds: seconds until the lock expires
+
+    :return: True on success or seconds left if already in use
+    """
+
+    # Remove expired locks first
+    Lock.objects.filter(until__lt=timezone.now()).delete()
+
+    # Use hash sum because this has a constant length
+    hash_sum = Lock.get_hash_sum(identifier)
+    obj, created = Lock.objects.get_or_create(hash_sum=hash_sum, defaults={
+        'duration': seconds,
+        'until': timezone.now() + timedelta(seconds=seconds)})
+
+    if created:
+        return True
```

Comment:
One more thing which might be of interest here. Not sure about the possibilities with the `@api_function`, but I would also use the response status to clarify what happened.
E.g. for the creation a response status `201` would be interesting and if the lock is already locked, there is `423` which is exactly `Locked`, which would fit perfectly here.

So, maybe you want to change the status codes too. Then the user can directly check for the status code instead of trying to interpret the content.

Reply:
I like the general idea but unfortunately this does not make sense for the Python Adminapi as we use it right now because it is built in a way that users do not have to deal with the HTTP response itself. 

We could on the long term think about providing a general HTTP API which supports that.